Skeleton/src/Volunteer.cpp

```diff
@@ -1 +1,263 @@
-#include "../include/Volunteer.h";
+
+#include "../include/Volunteer.h"
+#include "../include/Order.h"
+
+
+/////////////////////////////////////////////////////////
+//////////////////////// Volunteer //////////////////////
+/////////////////////////////////////////////////////////
+
+Volunteer::Volunteer(int id, const string &name) : id(id), name(name) 
+{
+    completedOrderId = -1;
+    activeOrderId = -1;
+}
+
+Volunteer::Volunteer(int id, const string &name, int completedOrderId, int activeOrderId) : id(id), name(name), completedOrderId(completedOrderId), activeOrderId(activeOrderId) {}
+
+int Volunteer::getId() const
+{
+    return id;
+}
+
+const string& Volunteer::getName() const 
+{
+    return name;
+}
+
+int Volunteer::getActiveOrderId() const 
+{
+    return activeOrderId;
+}
+
+bool Volunteer::isBusy() const
+{
+    return (activeOrderId != NO_ORDER);
+}
+
+
+/////////////////////////////////////////////////////////////////////////////////
+//////////////////////////// Collector Volunteer ////////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+
+
+CollectorVolunteer::CollectorVolunteer(int id, const string &name, const int coolDown) : Volunteer(id, name) ,coolDown(coolDown), timeLeft(NO_ORDER){}
+CollectorVolunteer* CollectorVolunteer::clone() const 
+{
+     return new CollectorVolunteer(*this);
+}
+void CollectorVolunteer::step() // need to check if it's correct
+{
+    if(decreaseCoolDown())
+    {
+        completedOrderId = activeOrderId;
+        activeOrderId = NO_ORDER;
+    }
+}
+int CollectorVolunteer::getCoolDown() const
+{
+    return coolDown;
+}
+
+int CollectorVolunteer::getTimeLeft() const 
+{
+    return timeLeft;
+}
+
+bool CollectorVolunteer::decreaseCoolDown() //Decrease timeLeft by 1,return true if timeLeft=0,false otherwise
+{
+    if(timeLeft > 0){
+        timeLeft = timeLeft - 1;
+    }
+    return (timeLeft == 0);
+}
+
+bool CollectorVolunteer::hasOrdersLeft() const  // Signal whether the volunteer didn't reach orders limit,Always true for CollectorVolunteer and DriverVolunteer
+{
+    return true;
+} 
+
+bool CollectorVolunteer::canTakeOrder(const Order &order) const 
+{
+    return (order.getStatus() == OrderStatus::PENDING && !Volunteer::isBusy());
+}
+
+void CollectorVolunteer::acceptOrder(const Order &order) 
+{
+    if(canTakeOrder(order)){
+        activeOrderId = order.getId();
+        timeLeft = coolDown;
+    }
+}
+string CollectorVolunteer::toString() const 
+{
+    return "Collector Volunteer - ID: " + std::to_string(getId())
```

Comment:
you have access to private so no getter needed

Reply:
We don't have access to private fields from the derived class
![image](https://github.com/OfekYabo/SPL-1/assets/130248635/a186bce3-a07d-4757-bce0-572f3c1afe8c)

Reply:
only if you have, this is not base class so maybe you dont have